### PR TITLE
Remove core addons from addons

### PR DESCRIPTION
**Pull request by hakman**
This "fake" addon is no longer needed.

/cc @olemarkus @rifelpet

### CODE REVIEW

**Comment by olemarkus** [comment]
There are a lot of addons causing the failure ofpull-kops-e2e-aws-upgrade-123-ko123-to-klatest-kolatest so you probably want to dodge this e2e here.

**Comment by hakman** [comment]
Let's see how things look without the removal of CCM leader migration:
```
channel_version.go:121] cluster has different ids for "leader-migration.rbac.addons.k8s.io" ("k8s-1.25" vs "k8s-1.23"); will replace
addon.go:169] Applying update from "s3://k8s-kops-prow/e2e-1e48ee0012-ddcd4.test-cncf-aws.k8s.io/addons/leader-migration.rbac.addons.k8s.io/k8s-1.25.yaml"
s3fs.go:329] Reading file "s3://k8s-kops-prow/e2e-1e48ee0012-ddcd4.test-cncf-aws.k8s.io/addons/leader-migration.rbac.addons.k8s.io/k8s-1.25.yaml"
apply.go:60] Running command: kubectl apply -f /tmp/channel2568152125/manifest.yaml --server-side --force-conflicts --field-manager=kops
apply.go:63] error running kubectl apply -f /tmp/channel2568152125/manifest.yaml --server-side --force-conflicts --field-manager=kops
apply.go:64] error: no objects passed to apply
```

**Comment by hakman** [comment]
@rifelpet Any quick idea why `[sig-node] SSH should SSH to all nodes and run command` fails consistently during the upgrade test?

**Comment by hakman** [comment]
I think it is expected to use Ubuntu 20.04 for now everywhere. I don't think we changed to 22.04 except for some experiments. If we broke something with ed25519 it should be pretty obvious.

What do you think about this change during upgrade, maybe the key gets replaced or broken?
```
  SSHKey/kubernetes.e2e-1e48ee0012-ddcd4.test-cncf-aws.k8s.io-5c:be:7c:c6:da:2e:d3:46:dd:0f:a1:84:f2:15:62:1c
  	Shared              	 false -> false
  	PublicKey           	 <nil> -> <resource>
  	KeyFingerprint      	 4KJtTE/1xAEOcBdOWMf1jvFAkvJC74BC0kVq6GCXdj4= -> SHA256:4KJtTE/1xAEOcBdOWMf1jvFAkvJC74BC0kVq6GCXdj4
```

**Comment by rifelpet** [comment]
The `=` being removed from the end of the fingerprint is very suspicious, as is the `PublicKey` field changing. There haven't been [any recent changes to the sshkey task](https://github.com/kubernetes/kops/blob/master/upup/pkg/fi/cloudup/awstasks/sshkey.go), nor its [model](https://github.com/kubernetes/kops/blob/master/pkg/model/awsmodel/sshkey.go). Maybe something related to the ssh key resources in the state store?

**Comment by hakman** [comment]
Tested and the above changes are not changing the key or affecting access to my test cluster.
The spurious changes will be removed in https://github.com/kubernetes/kops/pull/13774.

**Comment by hakman** [comment]
@rifelpet My only remaining idea is that [KUBE](https://github.com/kubernetes/kops/blob/3242dc3cb421e43865ebe6a19c64783ad2ecf423/tests/e2e/kubetest2-kops/deployer/common.go#L191) is not set when `kubetest2` doesn't deploy and the SSH test falls back to [$HOME/.ssh/kube_aws_rsa](https://github.com/kubernetes/kubernetes/blob/25fa92cd7d2359710d5e7e223ea5be3b69c6c7fa/test/e2e/framework/ssh/ssh.go#L77). In any case, it's not the key we need.

**Comment by rifelpet** [comment]
You're right, we don't seem to be setting `KUBE_SSH_KEY_PATH` when we execute ginkgo. I'm not sure how this succeeds in other e2e jobs

File: upup/pkg/fi/cloudup/bootstrapchannelbuilder/bootstrapchannelbuilder.go [hunk 1]
```diff
@@ -463,18 +451,6 @@ func (b *BootstrapChannelBuilder) buildAddons(c *fi.ModelBuilderContext) (*Addon
 				Id:       id,
 			})
 		}
-
-		if b.IsKubernetesGTE("1.25") {
```
[the remark this answers is not in the document]
Comment (hakman) [reply upup/pkg/fi/cloudup/bootstrapchannelbuilder/bootstrapchannelbuilder.go:467]: Done. 👍

**Review by olemarkus — APPROVED** [review]

**Pull request merged**